index.src.html

```diff
@@ -2208,90 +2246,115 @@
-    {{VideoFrameInit/codedHeight}}, return `false`.
-4. If {{VideoFrameInit/cropLeft}} + {{VideoFrameInit/cropWidth}} >=
-    {{VideoFrameInit/codedWidth}}, return `false`.
-5. If {{VideoFrameInit/displayWidth}} = 0 or
-    {{VideoFrameInit/displayHeight}} = 0, return `false`.
-6. Return `true`.
-
+: To check if a {{VideoFramePlanesInit}} is a
+    <dfn>valid VideoFramePlanesInit</dfn>, run these steps:
+:: 1. If {{VideoFramePlanesInit/codedWidth}} = 0 or
+        {{VideoFramePlanesInit/codedHeight}} = 0,return `false`.
+    2. If {{VideoFramePlanesInit/cropWidth}} = 0 or
+        {{VideoFramePlanesInit/cropHeight}} = 0, return `false`.
+    3. If {{VideoFramePlanesInit/cropTop}} +
+        {{VideoFramePlanesInit/cropHeight}} >=
+        {{VideoFramePlanesInit/codedHeight}}, return `false`.
+    4. If {{VideoFramePlanesInit/cropLeft}} +
+        {{VideoFramePlanesInit/cropWidth}} >=
+        {{VideoFramePlanesInit/codedWidth}}, return `false`.
+    5. If {{VideoFramePlanesInit/displayWidth}} = 0 or
+        {{VideoFramePlanesInit/displayHeight}} = 0, return `false`.
+    6. Return `true`.
+
+: <dfn>Clone VideoFrame</dfn> (with |frame|)
+:: 1. Let |clone| be a new {{VideoFrame}} initialized as follows:
+        1. Assign |frame|.{{VideoFrame/[[resource reference]]}} to
+            {{AudioFrame/[[resource reference]]}}.
+        2. Assign |frame|.{{VideoFrame/format}} to {{VideoFrame/format}}.
+        3. Assign a new [=list=] to {{VideoFrame/planes}}.
+        4. For each |plane| in {{VideoFrame/planes}}:
+            1. Let |clonePlane| be a new {{Plane}} initialized as follows:
+                1. Assign |clone| to |clonePlane|.{{Plane/[[parent frame]]}}.
+                2. Assign |plane|.{{Plane/stride}} to {{Plane/stride}}.
+                3. Assign |plane|.{{Plane/rows}} to {{Plane/rows}}.
+                4. Assign |plane|.{{Plane/length}} to {{Plane/length}}.
+            2. Append |clonePlane| to {{VideoFrame/planes}}.
+        5. Assign |frame|.{{VideoFrame/codedWidth}} to
+            {{VideoFrame/codedWidth}}.
+        6. Assign |frame|.{{VideoFrame/codedHeight}} to
+            {{VideoFrame/codedHeight}}.
+        7. Assign |frame|.{{VideoFrame/cropLeft}} to {{VideoFrame/cropLeft}}.
+        8. Assign |frame|.{{VideoFrame/cropTop}} to {{VideoFrame/cropTop}}.
+        9. Assign |frame|.{{VideoFrame/cropWidth}} to {{VideoFrame/cropWidth}}.
+        10. Assign |frame|.{{VideoFrame/cropHeight}} to
+            {{VideoFrame/cropHeight}}.
+        11. Assign |frame|.{{VideoFrame/codedHeight}} to
+            {{VideoFrame/codedHeight}}.
+        12. Assign |frame|.{{VideoFrame/displayWidth}} to
+            {{VideoFrame/displayWidth}}.
+        13. Assign |frame|.{{VideoFrame/displayHeight}} to
+            {{VideoFrame/displayHeight}}.
+        14. Assign |frame|.{{VideoFrame/duration}} to {{VideoFrame/duration}}.
+        15. Assign |frame|.{{VideoFrame/timestamp}} to
+            {{VideoFrame/timestamp}}.
+    2. Return |clone|.
```

Comment:
This style is overly verbose. It would be clearer to say to copy all attributes to the new `clone` except `x` and `y` and then spell out the steps for those.

Reply:
Done.